modules/openapi-generator/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -244,10 +244,34 @@ class ObjectSerializer
             $data = is_string($data) ? json_decode($data) : $data;
             $subClass = substr($class, 0, -2);
             $values = [];
+            if (!is_array($data)) {
+                throw new \InvalidArgumentException('expected array');
+            }
             foreach ($data as $key => $value) {
                 $values[] = self::deserialize($value, $subClass, null);
             }
             return $values;
+        } elseif (substr($class, 0, strlen('oneOf[')) === 'oneOf[') {
+          $inner = substr($class, strlen('oneOf['), -1);
+          $options = explode(',', $inner);
+          $deserialized = [];
+          $errors = [];
+          foreach ($options as $option) {
+            try {
+              $deserialized[] = ['option' => trim($option), 'value' => self::deserialize($data, trim($option), $httpHeaders)];
```

Comment:
the deserializer needs to start to throw invalidargumentexceptions to help pick the right model to in the end.